example/flutter_example/lib/main.dart

```diff
@@ -22,7 +22,31 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
-  Ros ros = Ros(url: 'ws://10.0.2.2:9090');
+  Ros ros;
+  Topic chatter;
+
+  @override
+  void initState() {
+    ros = Ros(url: 'ws://10.0.2.2:9090');
+    chatter = Topic(
+        ros: ros, name: '/chatter', type: "std_msgs/String", reconnectOnClose: true, queueLength: 10, queueSize: 10);
+    this.initConnection();
+    super.initState();
+  }
+
+  void initConnection() {
+    setState(() {
+      ros.connect();
+      chatter.subscribe();
+    });
+  }
+
+  void destroyConnection() async {
```

Comment:
Kinda similar to `initConnection`, you marked this `async` but aren't waiting for `chatter.unsubscribe()` to complete.
Again, not really an issue, just a tad confusing :)

Reply:
At one point I was awaiting, but for some reason got rid of it. I'll put the await back in.